op-chain-ops/cmd/celo-migrate/genesis.go

```diff
@@ -1,130 +1,649 @@
+
+var celoL1GenesisAllocJSON = map[uint64]string{
+	MainnetNetworkID:   mainnetAllocJSON,
+	AlfajoresNetworkID: alfajoresAllocJSON,
+	BaklavaNetworkID:   baklavaAllocJSON,
+}
+
+// GetCeloL1GenesisAlloc returns the legacy Celo L1 genesis allocation JSON for the given network ID.
+func GetCeloL1GenesisAlloc(config *params.ChainConfig) ([]byte, error) {
+	chainID := config.ChainID.Uint64()
+	allocJSON, ok := celoL1GenesisAllocJSON[chainID]
+	if !ok {
+		return nil, fmt.Errorf("no genesis allocation JSON found for network ID %d", chainID)
+	}
+	return []byte(allocJSON), nil
+}
 
-func AlfajoresGenesisBlockFromConfig(config *params.ChainConfig) *core.Genesis {
-	alfajoresAlloc := &types.GenesisAlloc{}
-	alfajoresAlloc.UnmarshalJSON([]byte(alfajoresAllocJSON))
+// BuildGenesis creates a genesis block from the given parameters.
+func BuildGenesis(config *params.ChainConfig, allocs, extraData []byte, timestamp uint64) *core.Genesis {
+	genesisAlloc := &types.GenesisAlloc{}
+	genesisAlloc.UnmarshalJSON(allocs)
 	return &core.Genesis{
 		Config:    config,
-		Timestamp: 0x5b843511,
-		ExtraData: hexutil.MustDecode(alfajoresExtraData),
-		Alloc:     *alfajoresAlloc,
+		Timestamp: timestamp,
+		ExtraData: extraData,
```

Comment:
What about this comment: https://github.com/celo-org/optimism/pull/167#discussion_r1655041771 ?

Reply:
Yep! [Here](https://github.com/celo-org/optimism/blob/d970855ec252dbc64022840169e4e0c3b0c378a9/op-chain-ops/cmd/celo-migrate/state.go#L365) I read the migrated genesis header from the db and pass in the `extraData` from that